src/minidump.rs

```diff
@@ -879,6 +1059,119 @@ impl<'a> MinidumpStream<'a> for MinidumpModuleList {
     }
 }
 
+/// An iterator over `MinidumpModule`s.
+#[allow(missing_debug_implementations)]
+pub struct UnloadedModules<'a> {
+    iter: Box<dyn Iterator<Item = &'a MinidumpUnloadedModule> + 'a>,
+}
+
+impl<'a> Iterator for UnloadedModules<'a> {
+    type Item = &'a MinidumpUnloadedModule;
+
+    fn next(&mut self) -> Option<&'a MinidumpUnloadedModule> {
+        self.iter.next()
+    }
+}
+
+impl MinidumpUnloadedModuleList {
+    /// Return an empty `MinidumpModuleList`.
+    pub fn new() -> MinidumpUnloadedModuleList {
+        MinidumpUnloadedModuleList {
+            modules: vec![],
+            modules_by_addr: RangeMap::new(),
+        }
+    }
+    /// Create a `MinidumpModuleList` from a list of `MinidumpModule`s.
+    pub fn from_modules(modules: Vec<MinidumpUnloadedModule>) -> MinidumpUnloadedModuleList {
+        let modules_by_addr = modules
+            .iter()
+            .enumerate()
+            .map(|(i, module)| (module.memory_range(), i))
+            .into_rangemap_safe();
+        MinidumpUnloadedModuleList {
+            modules,
+            modules_by_addr,
+        }
+    }
+
+    /// Return a `MinidumpUnloadedModule` whose address range covers `address`.
+    pub fn module_at_address(&self, address: u64) -> Option<&MinidumpUnloadedModule> {
+        self.modules_by_addr
+            .get(address)
+            .map(|&index| &self.modules[index])
+    }
+
+    /// Iterate over the modules in arbitrary order.
+    pub fn iter(&self) -> UnloadedModules<'_> {
+        UnloadedModules {
+            iter: Box::new(self.modules.iter()),
```

Comment:
Yeah I'm not a Fan of this but it's pre-existing from the ModulesList, and if it shows up in profiles I think it's justified to make a breaking change here and split up the iterators.

Reply:
FWIW, the existing code here predates `impl trait`, so feel free to use that (and even change the other iterator-returning parts of the API, if you're feeling frisky)!

Reply:
`impl trait` has the downside users can't easily store it in a struct afaik, i thought in general people avoided it in public APIs because of this.  but yes, can be changed if it becomes a problem

Reply:
It's definitely more annoying to store now, but it's more honest about how much effort we're putting into the entry point.